### PR TITLE
Explicitly set the minimum supported rust version

### PR DESCRIPTION
**Pull request by SamWilsn**
Depends on gakonst/evm#1. Probably need to update the lockfile before merging.

Besides setting the Rust version:
 - Run the tests against 1.51 plus stable.
 - Run the tests against the whole workspace.
 - Run clippy against the whole workspace.
 - Fix a couple clippy warnings.
 - Add `--from-utf8` and make `--from-ascii` an alias to it, since UTF-8 is more accurate.

### CODE REVIEW

File: dapp/src/solc.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ impl<'a> SolcBuilder<'a> {
             .join(",");
 
         // tracing::trace!(?lib_paths);
-        solc = solc.args(["--allow-paths", &lib_paths]);
+        solc = solc.args(std::array::IntoIter::new(["--allow-paths", &lib_paths]));
```
Comment (gakonst) [inline dapp/src/solc.rs:78]: Why is this change needed?

File: dapptools/src/seth.rs [hunk 2]
```diff
@@ -16,17 +16,17 @@ use structopt::StructOpt;
 async fn main() -> eyre::Result<()> {
     let opts = Opts::from_args();
     match opts.sub {
-        Subcommands::FromAscii { text } => {
-            println!("{}", SimpleSeth::from_ascii(&text));
+        Subcommands::FromUtf8 { text } => {
```
Comment (gakonst) [inline dapptools/src/seth.rs:19]: We want the ascii aliases to be compatible with upstream! Let's have both names with the same implementation and note that one is the alias of the other?

File: seth/src/lib.rs [hunk 3]
```diff
@@ -199,15 +199,15 @@ where
...
 pub struct SimpleSeth;
 impl SimpleSeth {
-    /// Converts ASCII text input to hex
+    /// Converts UTF-8 text input to hex
     ///
     /// ```
     /// use seth::SimpleSeth as Seth;
     ///
-    /// let bin = Seth::from_ascii("yo");
+    /// let bin = Seth::from_utf8("yo");
```
Comment (gakonst) [inline seth/src/lib.rs:207]: Let's keep both and make one the alias of the other

File: seth/src/lib.rs [hunk 4]
```diff
@@ -248,18 +248,18 @@ impl SimpleSeth {
...
     /// assert_eq!(bytes, "0x1234000000000000000000000000000000000000000000000000000000000000");
     ///
-    /// let err = Seth::to_bytes32("0x123400000000000000000000000000000000000000000000000000000000000011").unwrap_err();
+    /// let err = Seth::bytes32("0x123400000000000000000000000000000000000000000000000000000000000011").unwrap_err();
     /// assert_eq!(err.to_string(), "string >32 bytes");
     ///
     /// # Ok(())
     /// # }
-    pub fn to_bytes32(s: &str) -> Result<String> {
+    pub fn bytes32(s: &str) -> Result<String> {
```
Comment (gakonst) [inline seth/src/lib.rs:262]: Let's keep the old name for upstream dapptools compatibility

**Review by gakonst — CHANGES_REQUESTED** [review]

**Comment by gakonst** [comment]
Really nice work! Some nits about compatibility with upstream but lgtm once these are addressed.

File: dapp/src/solc.rs (hunk 1 above)
Comment (SamWilsn) [reply dapp/src/solc.rs:78]: The iterator impl that you were using didn't exist in 1.51

File: dapptools/src/seth.rs (hunk 2 above)
Comment (SamWilsn) [reply dapptools/src/seth.rs:19]: Upstream being the original command line tool? I added an alias for that here: https://github.com/gakonst/dapptools-rs/pull/28/files#diff-162c4eb7d539d0bf957db975fd66a58d2fff454e7bc8fca1c3f78733e02f465fR14

Or did you mean from an API/ABI point of view?

**Comment by SamWilsn** [comment]
> Really nice work! Some nits about compatibility with upstream but lgtm once these are addressed. I merged the PR on the EVM, can you update the dep to that?

Can do!

> Also, might be worth up streaming?

The `evm` crate changes? I can certainly try!

File: dapptools/src/seth.rs (hunk 2 above)
Comment (gakonst) [reply dapptools/src/seth.rs:19]: Oh sorry my mistake, I did not see the `aliases` tag.

**Review by gakonst — APPROVED** [review]
